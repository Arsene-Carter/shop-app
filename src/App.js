import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h2>Shopping List App</h2>
        { 
          this.props.children
        }
      </div>
    );
  }
}

export default App;
