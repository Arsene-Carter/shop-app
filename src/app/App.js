// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions';
import { styled } from 'styletron-react';
import { getSessionUserId } from '../reducers';
import { Button } from '../ui';

type Props = {
  children: {},
  logout: () => void,
  haveSession: boolean,
  isLoading: boolean,
};

class App extends Component {
  props: Props;

  getContent() {
    return (
      <AppContainer>
        <h2>Shopping List App</h2>
        {
          this.props.children
        }
      </AppContainer>
    );
  }

  render() {
    return (
      <div>
        <AppHeader>
          <Flex1 />
          <Button
            onClick={this.props.logout}
          >
            Logout
          </Button>
        </AppHeader>
        { this.props.isLoading ? 'Loading... ' : this.getContent() }
      </div>
    );
  }
}

const AppContainer = styled('div', {
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
});

const Flex1 = styled('div', {
  flex: 1,
});
const AppHeader = styled('div', {
  display: 'flex',
  flexDirection: 'row',
  paddingRight: '10px',
});

const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(actions.logoutRequest()),
});

const mapStateToProps = state => ({
  haveSession: getSessionUserId(state) != null,
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
