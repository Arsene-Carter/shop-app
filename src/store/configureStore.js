// @flow
import { createStore, applyMiddleware, compose } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';
import { authMiddleware } from '../middlewares';
import { mainEpic } from '../epics';

const composeEnhancers =
    process.env.NODE_ENV !== 'production' &&
    typeof window === 'object' &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?   
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        // Specify here name, actionsBlacklist, actionsCreators and other options
      }) : compose;

const epicMiddleware = createEpicMiddleware(mainEpic);

const enhancer = composeEnhancers(
    applyMiddleware(thunk, epicMiddleware),
  );


const configureStore = preloadedState => createStore(
    rootReducer,
    preloadedState,
    enhancer,
);

export default configureStore;
