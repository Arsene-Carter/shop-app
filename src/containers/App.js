// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as sessionActions from '../store/session/actions';
import { styled } from 'styletron-react';
import { getSessionUserId } from '../store/mainReducer';
import { Button } from '../ui';
import { LayoutVertical } from '../ui/layout';

type Props = {
  children: {},
  logout: () => void,
  haveSession: boolean,
  isLoading: boolean,
};

class App extends Component {
  props: Props;

  getContent() {
    return (
      <AppContainer>
        <h2>Shopping List App</h2>
        {
          this.props.children
        }
      </AppContainer>
    );
  }

  render() {
    return (
      <LayoutVertical height="100vh">
        <AppHeader>
          <Flex1 />
          <Button
            onClick={this.props.logout}
          >
            Logout
          </Button>
        </AppHeader>
        { this.props.isLoading ? 'Loading... ' : this.getContent() }
      </LayoutVertical>
    );
  }
}

const AppContainer = styled('div', {
  display: 'flex',
  flex: 1,
  flexDirection: 'column',
  alignItems: 'center',
});

const Flex1 = styled('div', {
  flex: 1,
});
const AppHeader = styled('div', {
  display: 'flex',
  flexDirection: 'row',
  paddingRight: '10px',
});

const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(sessionActions.logoutRequest()),
});

const mapStateToProps = state => ({
  haveSession: getSessionUserId(state) != null,
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
