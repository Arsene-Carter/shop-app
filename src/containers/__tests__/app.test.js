import renderer from 'react-test-renderer';
import React from 'react';
import App from '../App';
import configureStore from '../../store/configureStore';
import WithStyletron from '../../shared/styletron/configure';


it('render  App', () => {
  const appTree = renderer.create(
    <WithStyletron>
      <App store={configureStore()} />
    </WithStyletron>
  );
  expect(appTree).toMatchSnapshot();
});
