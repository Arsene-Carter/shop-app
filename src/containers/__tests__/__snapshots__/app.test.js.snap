exports[`test render  App 1`] = `
<div
  className="a b c d e f g h i j k l"
  height="100vh">
  <div
    className="a b c d e m n o f g h i j p q r s">
    <div
      className="t m n o a b c d e f g h i j u">
      <nav
        className={undefined}>
        <nav
          className="">
          <nav
            className="">
            <a
              className="">
              <svg
                height="24px"
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 1024 1024"
                width="24px">
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  style={
                    Object {
                      "fill": "#7a7a7a",
                    }
                  } />
              </svg>
            </a>
          </nav>
        </nav>
      </nav>
    </div>
  </div>
</div>
`;
